### PR TITLE
APA102 LED Library

### PR DESCRIPTION
**Pull request by HappyCodingRobot**
This PR is a basic library for the **APA102** LED controller. (aka SuperLED, Adafruit DotStar, ..)
It is a RGB LED controller chip with data_in & clk Signal and can be used with SPI mode 3.

### CODE REVIEW

**Comment by hreintke** [comment]
@HappyCodingRobot @harry-boe : 
Is the part of this PR which creates base class for SPI handled by the generic SPI activities from @harry-boe ?

**Comment by hreintke** [comment]
@alonewolfx2 : 
Can you take a look at this library and comment ?

**Comment by hreintke** [comment]
@alonewolfx2 : 
Can you have a look at this implementation ? 
@HappyCodingRobot : If @alonewolfx2 agrees with this, can you squash the commits before getting it merged ?

File: Sming/Libraries/ArduCAM/ArduCAM.cpp [hunk 1]
```diff
@@ -0,0 +1,1028 @@
...
+	const struct sensor_reg *next = reglist;
+	
+	while ((reg_addr != 0xff) | (reg_val != 0xff))
+	{		
+		reg_addr = pgm_read_word(&next->reg);
+		reg_val = pgm_read_word(&next->val);
+		err = wrSensorReg8_8(reg_addr, reg_val);
+   	next++;
+#if defined(ESP8266)
+	  yield();
```
Comment (slaff) [inline Sming/Libraries/ArduCAM/ArduCAM.cpp]: Where is `yield` coming from? Was that code compiled with the ESP8266 directive ?

File: samples/Humidity_DHT22/app/application.cpp [hunk 2]
```diff
@@ -47,9 +47,9 @@ void init()
 	if(dht.readTempAndHumidity(th))
 	{
 		Serial.print("\tHumidity: ");
-		Serial.print(h);
+		Serial.print(th.humid);
```
Comment (slaff) [inline samples/Humidity_DHT22/app/application.cpp]: Please, split the PR into smaller PRs. I can see that there are changes related to:
* APA 102 LED
* ArduCAM
* DHT22
etc..

**Comment by slaff** [comment]
@HappyCodingRobot This PR https://github.com/SmingHub/Sming/pull/787 contains all changes related to APA 201 LED.

**Pull request closed without merging**